src/Autodesk.Forge.Core/ForgeHandler.cs

```diff
@@ -74,7 +81,7 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                 // add a retry policy so that we refresh invalid tokens
                 policies = policies.WrapAsync(GetTokenRefreshPolicy());
             }
-            return await policies.ExecuteAsync(async () => await base.SendAsync(request, cancellationToken));
+            return await policies.ExecuteAsync(async (ct) => await base.SendAsync(request, ct), cancellationToken);
```

Comment:
This is an extremely important change. It allows the Timeout policy to cancel the request.

Reply:
sometimes I am still confused how the cancel would work.. so supposing a request hit 15s timeout limit (the current one), we will get exception, but we will do retry then(line 113 below), then when the cancelling of the request will happen?

Reply:
If a request goes for more than 15s. Our timeout policy will cancel it and throw a TimeoutRejectedException. Our retry policy will then retry the request up to 5 times (with exponential backoff and jitter). If all 5 requests fail then ultimately a TimeoutRejectedException will propagate. See https://github.com/Autodesk-Forge/forge-api-dotnet-core/blob/9564b979a69e6894ce66f0424d4a5cbd92c135c5/tests/Autodesk.Forge.Core.Test/TestForgeHandler.cs#L411

Reply:
ok, but why we can see `Polly.Timeout.TimeoutRejectedException` exception now(the log I post in fda-ops channel) even without cancellation token enabled?

Reply:
never mind, I am reading the document here:https://github.com/App-vNext/Polly/wiki/Timeout, realizing this is a difference between Pessimistic timeout and Optimistic timeout!